src/cmd/windows/durable_task_monitor.go

```diff
@@ -0,0 +1,160 @@
+		common.RecordExit(-2, resultPath, scriptLogger)
+		return
+	}
+	pid := scriptCmd.Process.Pid
+	launchLogger.Printf("launched %v\n", pid)
+	err = scriptCmd.Wait()
+	common.CheckIfErr(scriptLogger, err)
+	resultVal := scriptCmd.ProcessState.ExitCode()
+	launchLogger.Printf("script exit code: %v\n", resultVal)
+
+	common.RecordExit(resultVal, resultPath, launchLogger)
+}
+
+func main() {
+	var controlDir, resultPath, logPath, executable, args, outputPath string
+	var debug, daemon bool
+	const controlFlag = "controldir"
+	const resultFlag = "result"
+	const logFlag = "log"
+	const executableFlag = "executable"
+	const argsFlag = "args"
+	const outputFlag = "output"
+	const debugFlag = "debug"
+	const daemonFlag = "daemon"
+	flag.StringVar(&controlDir, controlFlag, "", "working directory")
+	flag.StringVar(&resultPath, resultFlag, "", "full path of the result file")
+	flag.StringVar(&logPath, logFlag, "", "full path of the log file")
+	flag.StringVar(&executable, executableFlag, "", "path to the executable being launched")
+	flag.StringVar(&args, argsFlag, "", "(optional) argument(s) to the executable, separated by commas WITH spaces")
+	flag.StringVar(&outputPath, outputFlag, "", "(optional) if recording output, full path of the output file")
+	flag.BoolVar(&debug, debugFlag, false, "(optional) noisy output to log")
+	flag.BoolVar(&daemon, daemonFlag, false, "(optional) Free binary from parent process")
+	flag.Parse()
+
+	// Validate that the required flags were all command-line defined
+	required := []string{controlFlag, resultFlag, logFlag, executableFlag}
+	defined := make(map[string]string)
+	flag.Visit(func(f *flag.Flag) {
+		defined[f.Name] = f.Value.String()
+	})
+	if !common.ValidateFlags(defined, required) {
+		os.Exit(-2)
+	}
+
+	fmt.Fprintf(os.Stdout, "Parent pid is: %v\n", os.Getppid())
+
+	if daemon {
+		fmt.Fprintf(os.Stdout, "1st launch pid is: %v\n", os.Getpid())
+		rebuiltArgs := common.RebuildArgs(defined, daemonFlag)
+		doubleLaunchCmd := exec.Command(os.Args[0], rebuiltArgs...)
+		doubleLaunchCmd.Stdin = nil
+		doubleLaunchCmd.Stdout = nil
+		doubleLaunchCmd.Stderr = nil
+		var sysAttr windows.SysProcAttr
+		sysAttr.HideWindow = true
+		sysAttr.CreationFlags = windows.CREATE_NEW_PROCESS_GROUP
+		sysAttr.NoInheritHandles = true
+		doubleLaunchCmd.SysProcAttr = &sysAttr
+		doubleLaunchErr := doubleLaunchCmd.Start()
+		if doubleLaunchErr != nil {
+			panic("Double launch failed, exiting")
+		}
+		return
+	}
+	// Prepare logging
+	logFile, logErr := os.Create(logPath)
+	if logErr != nil {
+		fmt.Fprintf(os.Stderr, "Unable to create log file: %s", logErr)
+		common.RecordExit(-2, resultPath, log.Default())
+		return
+	}
+	defer logFile.Close()
+	mainLogger, hbLogger, launchLogger, scriptLogger := common.PrepareLogging(logFile, debug)
+
+	for key, val := range defined {
+		mainLogger.Printf("%v: %v", key, val)
+	}
+	mainLogger.Printf("Main pid is: %v\n", os.Getpid())
+	mainLogger.Printf("Parent pid is: %v\n", os.Getppid())
```

Comment:
Does this stuff show up by default in Jenkins, or only when `debug` is `true`?

Reply:
Only when `debug` is `true`